sdl_android/src/main/java/com/smartdevicelink/api/screen/ScreenManager.java

```diff
@@ -0,0 +1,343 @@
+package com.smartdevicelink.api.screen;
+
+import android.support.annotation.NonNull;
+import android.util.Log;
+
+import com.smartdevicelink.api.BaseSubManager;
+import com.smartdevicelink.api.CompletionListener;
+import com.smartdevicelink.api.FileManager;
+import com.smartdevicelink.api.datatypes.SdlArtwork;
+import com.smartdevicelink.proxy.interfaces.ISdl;
+import com.smartdevicelink.proxy.rpc.SetDisplayLayout;
+import com.smartdevicelink.proxy.rpc.enums.MetadataType;
+import com.smartdevicelink.proxy.rpc.enums.PredefinedLayout;
+import com.smartdevicelink.proxy.rpc.enums.TextAlignment;
+
+import java.util.List;
+
+/**
+ * <strong>ScreenManager</strong> <br>
+ *
+ * Note: This class must be accessed through the SdlManager. Do not instantiate it by itself. <br>
+*/
+public class ScreenManager extends BaseSubManager {
+
+	private static String TAG = "ScreenManager";
+	private FileManager fileManager;
+	private SoftButtonManager softButtonManager;
+	private TextAndGraphicManager textAndGraphicManager;
+	private boolean allSubManagersFinishedUpdatingSuccesslly;
+
+	// Sub manager listener
+	private final CompletionListener subManagerListener = new CompletionListener() {
+		@Override
+		public synchronized void onComplete(boolean success) {
+			if(!success){
+				Log.d(TAG, "Sub manager failed to initialize");
+			}
+			if(
+					softButtonManager != null && softButtonManager.getState() != BaseSubManager.SETTING_UP &&
+					textAndGraphicManager != null && textAndGraphicManager.getState() != BaseSubManager.SETTING_UP
+					){
+				transitionToState(READY);
+			}
+		}
+	};
+
+	public ScreenManager(ISdl internalInterface, FileManager fileManager) {
+		super(internalInterface);
+		transitionToState(SETTING_UP);
+		this.fileManager = fileManager;
+		initialize();
+	}
+
+	private void initialize(){
+		this.softButtonManager = new SoftButtonManager(internalInterface, fileManager);
+		this.softButtonManager.start(subManagerListener);
+		this.textAndGraphicManager = new TextAndGraphicManager(internalInterface, fileManager, softButtonManager);
+		this.textAndGraphicManager.start(subManagerListener);
+	}
+
+	/**
+	 * <p>Called when manager is being torn down</p>
+	 */
+	public void dispose(){
+		transitionToState(SHUTDOWN);
+	}
+
+	/**
+	 * Set the textField1 on the head unit screen
+	 * @param textField1 String value represents the textField1
+	 */
+	public void setTextField1(@NonNull String textField1) {
```

Comment:
Is there a reason this has the `@NonNull` tag while all the other fields do not? If so, it would be good to include the reason in the javadoc. Also including a note that empty set`""` will clear the field.